### PR TITLE
Feature/interviews

### PR DESCRIPTION
**Pull request by ralitsadineva**
Interview feedback, application status, statistics, search

### CODE REVIEW

File: services/positions.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def add(id, title, description):
         return {'success': False, 'error': {'error': True}}
 
 def duplicate(id):
-    return positions_db.get(id)[2:4]
+    return [positions_db.get(id)['title'], positions_db.get(id)['description']]
```
Comment (ilian-iliev-io) [inline services/positions.py]: position = positions_db.get(id)

File: services/users.py [hunk 2]
```diff
@@ -132,5 +132,5 @@ def update_company(id, company):
         return {'success': False, 'error': {'error': True}}
 
 def get_signature_company(id):
-    signature, company = users_db.get(id)[8:10]
+    signature, company = users_db.get(id)['signature'], users_db.get(id)['company']
```
Comment (ilian-iliev-io) [inline services/users.py]: .

File: templates/candidate.html [hunk 3]
```diff
@@ -202,5 +202,113 @@
...
+                </select>
+            </form>
+        {% endif %}
+        {% if candidate['reject_reason'] %}
+            <div class="border-bottom border-success-subtle">
+                <a role="button" class="btn btn-success mt-2 mb-2" href="{{ base_url }}positions/{{ candidate['pos_id'] }}/{{ candidate['cand_id'] }}/unreject">Unmark rejected</a>
+            </div>
+        {% endif %}
+        {% if candidate['decline_reason'] %}
+            <p class="mt-3 fw-semibold border-top border-success-subtle">Declined because of {% if candidate['decline_reason'] == '1' %}the company culture or environment{% elif candidate['decline_reason'] == '2' %}the salary{% elif candidate['decline_reason'] == '3' %}the location{% elif candidate['decline_reason'] == '4' %}not enough career growth opportunities{% elif candidate['decline_reason'] == '5' %}the job responsibilities{% elif candidate['decline_reason'] == '6' %}work-life balance{% elif candidate['decline_reason'] == '7' %}lack of home office/working hours flexibility{% endif %}.</p>
```
Comment (ilian-iliev-io) [inline templates/candidate.html]: DECLINE_1 = '1'
DECLINE_2 = '2'
DECINE_REASONS = { DECLINE_1:  'test 1', DECLINE_2:  'test 2'} 

def get_decline_reason_string(reason):
     return DECINE_REASONS[reason]
     
  {% decline_reason_string %}

File: templates/candidate.html [hunk 4]
```diff
@@ -202,5 +202,113 @@
...
+        {% endif %}
+        {% if not candidate['hired'] %}
+            <form class="mt-2 input-group" method="POST">
+                <button type="submit" class="btn btn-success" name="form_decline_reason">Mark that they declined because of</button>
+                <select class="form-select" id="declineReason" name="decline_reason" required>
+                    <option selected disabled value="">Choose reason</option>
+                    <option value="1">the company culture or environment</option>
```
Comment (ilian-iliev-io) [inline templates/candidate.html]: dynamic generation

File: templates/interview.html [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+            <div class="mb-3 row">
+                <label for="inputScore" class="col-sm-3 col-form-label fw-semibold">Score</label>
+                <div class="col-sm-9">
+                    <div class="input-group">
+                        <div>
+                            <select class="form-select" id="inputScore" name="score" required>
+                                {% if not interview %}
+                                    <option selected disabled value="">-</option>
+                                {% endif %}
+                                <option {% if interview and interview['score']==1 %}selected{% endif %} value="1">1</option>
```
Comment (ilian-iliev-io) [inline templates/interview.html]: loop

**Review by ilian-iliev-io — CHANGES_REQUESTED** [review]

**Review by ilian-iliev-io — APPROVED** [review]

**Review by ilian6806 — APPROVED** [review]

**Pull request merged**